src/tools/rustfmt/tests/source/issue_3853.rs

```diff
@@ -50,3 +50,13 @@ if let Some(ref     /*def*/  mut     /*abc*/       state)=          /*abc*/foo{
 				println!(
         "asdfasdfasdf");	}
 }
+
+fn mut_ref_mut() {
+    if let Some(mut /*a*/ ref     /*def*/  mut     /*abc*/       state)=          /*abc*/foo{
+        println!(
+"asdfasdfasdf");	}
+
+if let Some(mut /*a*/ ref     /*def*/  /*mut*/       state)=          /*abc*/foo{
+    println!(
+"asdfasdfasdf");	}
+}
```

Comment:
nit: would you mind moving these tests into their own file and adding a `#![feature(mut_ref)]` annotation so it's easier to remember that these are associated with the new feature you're working on.